src/level_setup.m

```diff
@@ -0,0 +1,94 @@
+function level_collection = level_setup(directory)
+% build_level reads a directory containing level files. Level files are 
+% .txt files containing the objects to be placed in the
+% level. 
+
+% Input level_name = the name of the .txt file
+% Output level_collection = a Level array containing all levels in
+% directory.
+
+addpath(directory);
+D = dir(directory);
+ii=1;
+for nn = 1:length(D)
+    if D(nn).isdir
+        continue
+    else
+        [C,level_name] = read_level(D(nn));
+        level_collection(ii) = build_level(C,level_name);
+        ii = ii+1;
+    end
+    
+end
+
+end
+
+function [C,level_name] = read_level(D)
+level_name = D.name;
+fid = fopen(level_name);
+C = textscan(fid,'%s %f %f %f %f','headerLines',1,'TreatAsEmpty','none');
+fclose(fid);
+end
+
+function level_collection = build_level(C,level_name)
+lvl = Level();
+lvl.set_id(str2num(level_name(6)));
+j=1;
+for qq = 1:size(C{1},1)
```

Comment:
If possible, don't loop using index-based for loops, but range based: not `for i=1:len; some_array(i)` but `for element = some_array`. Also in other places.